tutor/src/components/change-student-id.cjsx

```diff
@@ -0,0 +1,104 @@
+_ = require 'lodash'
+React = require 'react'
+BS = require 'react-bootstrap'
+
+LoadableItem = require './loadable-item'
+{ ChangeStudentIdForm } = require 'shared'
+{ TransitionActions, TransitionStore } = require '../flux/transition'
+{ CourseStore, CourseActions } = require '../flux/course'
+{ StudentIdStore, StudentIdActions } = require '../flux/student-id'
+
+ERROR_MAP = {
+  student_identifier_has_already_been_taken:
+    'The provided student ID has already been used in this course. ' +
+    'Please try again or contact your instructor.'
+  blank_student_identifer:
+    'The student ID field cannot be left blank. Please enter your student ID.'
+}
+
+module.exports = React.createClass
+  contextTypes:
+    router: React.PropTypes.func
+
+  onCancel: -> @goBack()
+
+  componentWillMount: ->
+    {courseId} = @context.router.getCurrentParams()
+    CourseStore.on('course.loaded', @update)
+    CourseActions.load(courseId)
+
```

Comment:
I don't see CourseStore being unbound anywhere.  You probably don't need to load it though, the course data is already loaded by the bootstrap data.